LeaderboardBackend/Services/Impl/AccountConfirmationService.cs

```diff
@@ -0,0 +1,61 @@
+using LeaderboardBackend.Models.Entities;
+using NodaTime;
+
+namespace LeaderboardBackend.Services;
+
+public class AccountConfirmationService : IAccountConfirmationService
+{
+    private readonly ApplicationContext _applicationContext;
+    private readonly IEmailSender _emailSender;
+
+    public AccountConfirmationService(ApplicationContext applicationContext, IEmailSender emailSender)
+    {
+        _applicationContext = applicationContext;
+        _emailSender = emailSender;
+    }
+
+    public async Task<AccountConfirmation?> GetConfirmationById(Guid id)
+    {
+        return await _applicationContext.AccountConfirmations.FindAsync(id);
+    }
+
+    public async Task<CreateConfirmationResult> CreateConfirmationAndSendEmail(User user, CancellationToken token = default)
+    {
+        if (user.Role is not UserRole.Registered)
+        {
+            return new BadRole();
+        }
+
+        AccountConfirmation newConfirmation =
+            new()
+            {
+                CreatedAt = SystemClock.Instance.GetCurrentInstant(),
+                ExpiresAt = SystemClock.Instance.GetCurrentInstant() + Duration.FromHours(1),
```

Comment:
You could reduce this to one `GetCurrentInstant()` call.

Reply:
Oh yeah lol